### PR TITLE
SDV manager delivery

### PR DESCRIPTION
**Pull request by Mark-J-Lawrence**
This is the Pull Request to deliver the new **Security Definition Validation** manager which makes use of the new SDC functionality being delivered in CICS TS 6.2.

Related Galasa documentation for this manager will be delivered via https://github.com/galasa-dev/galasa.dev/pull/788

For the purpose of the docs, I've set this release as an `alpha` release due to the level of usage and testing its been through. Once it is in use by the CiP pipelines, and confidence in it increases, this could change.

The manager code has unit testing covering 84% lines of code, and the build is set up to fail when the coverage goes any lower than this. However, jacoco must be enabled, & as I've started using the `galasa.manager` plugin, I've noticed its disabled by default globally, which switches off the coverage checking. I have globally enabled it in this PR, which had no ill effects as part of the local build, but I don't know what else that might affect?

I have also introduced **checkstyle**, **PMD**, and **spotbugs** static analysis as part of the SDV manager build, which will fail if any rules are broken.

I have now tested that the manager works as part of this repository via this branch by cloning the `framework` ,` wrapping`, `buildutils`, `extensions`, and `obr` repositories to create a new `0.34.0` uber-OBR locally, which I can then use for my tests locally. The 0.34.0 version of the manager is picked up, and the tests pass.

I have written an IVT for the manager which I have not tested, because I don't know how, but, the SDV manager relies on some CPS properties being set, so I would fully expect the IVT to fail without the following properties being added to the ecosystem running the IVTs (I think these may be best to pass in as overrides via a file when the IVTs are started):

```
sdv.roleTag.role1.role=TEST
sdv.zosImage.<zos image tag>.role.TEST.credTags=<a credential tag>
sdv.cicsTag.<cics tag>.port=<an available port to set up SDC on>
sdv.cicsTag.<cics tag>.hlq=ANTZ.CICS.TS.DEV.INTEGRAT
sdv.cicsTag.<cics tag>.SdcActivation=true
sdv.cicsTag.<cics tag>.SrrLogstreamRemoval=true
```
The IVT will currently only pass when ran on a region running CICS TS 6.2 or greater.

### CODE REVIEW

**Comment by galasa-team** [comment]
Build failed, see http://localhost:8001/api/v1/namespaces/tekton-pipelines/services/tekton-dashboard:http/proxy/#/namespaces/galasa-build/pipelineruns/repo-managers-pr-940-v965d for details. If you are unable to do so, please contact a member of the Galasa team.

File: galasa-managers-parent/gradle.properties [hunk 1]
```diff
@@ -1,6 +1,6 @@
 org.gradle.jvmargs=-Xmx4096M
 
-jacocoEnabled= false
+jacocoEnabled= true
```
Comment (Mark-J-Lawrence) [inline galasa-managers-parent/gradle.properties:3]: I have changed this global setting to switch on jacoco, which may be controversial? 

I would like jacoco on regardless for the SDV manager due to the coverage threshold checking.

**Comment by galasa-team** [comment]
Build failed, see http://localhost:8001/api/v1/namespaces/tekton-pipelines/services/tekton-dashboard:http/proxy/#/namespaces/galasa-build/pipelineruns/repo-managers-pr-940-njs2s for details. If you are unable to do so, please contact a member of the Galasa team.

**Comment by galasa-team** [comment]
Build failed, see http://localhost:8001/api/v1/namespaces/tekton-pipelines/services/tekton-dashboard:http/proxy/#/namespaces/galasa-build/pipelineruns/repo-managers-pr-940-2dc4q for details. If you are unable to do so, please contact a member of the Galasa team.

**Comment by galasa-team** [comment]
Build failed, see http://localhost:8001/api/v1/namespaces/tekton-pipelines/services/tekton-dashboard:http/proxy/#/namespaces/galasa-build/pipelineruns/repo-managers-pr-940-hvxm8 for details. If you are unable to do so, please contact a member of the Galasa team.

**Comment by galasa-team** [comment]
Build successful

**Comment by galasa-team** [comment]
Build successful

**Comment by galasa-team** [comment]
Build successful

File: galasa-managers-parent/gradle.properties (hunk 1 above)
Comment (techcobweb) [reply galasa-managers-parent/gradle.properties:3]: I think that's a good idea too.

File: galasa-managers-parent/galasa-managers-testingtools-parent/dev.galasa.sdv.manager.ivt/build.gradle [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    implementation project (':galasa-managers-testingtools-parent:dev.galasa.sdv.manager')
+}
+
+// Note: These values are consumed by the parent build process
+// They indicate which packages of functionality this OSGi bundle should be delivered inside,
+// or referenced from.
+// The settings here are gathered together by the build process to create a release.yaml file 
+// which gathers-up all the packaging metadata about all the OSGi bundles in this component.
+ext.projectName=project.name
+ext.includeInOBR          = false
```
Comment (techcobweb) [inline galasa-managers-parent/galasa-managers-testingtools-parent/dev.galasa.sdv.manager.ivt/build.gradle:25]: You probably want to set the InOBR flag, so it gets included with all the other managers.

File: galasa-managers-parent/galasa-managers-testingtools-parent/dev.galasa.sdv.manager.ivt/build.gradle [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+ext.includeInMVP          = false
+ext.includeInBOM          = false
```
Comment (techcobweb) [inline galasa-managers-parent/galasa-managers-testingtools-parent/dev.galasa.sdv.manager.ivt/build.gradle:27]: And in the BOM

**Review by techcobweb — APPROVED** [review]

**Comment by galasa-team** [comment]
Build successful

**Comment by galasa-team** [comment]
Build successful

**Pull request merged**